src/keystore/persistence/EncryptedPersistence.ts

```diff
@@ -94,13 +41,35 @@ export default class EncryptedPersistence implements Persistence {
   }
 
   private async encrypt(value: Uint8Array): Promise<Uint8Array> {
-    const ecies = await eccrypto.encrypt(this.publicKey, Buffer.from(value))
-    return serializeEcies(ecies)
+    const ecies = await encrypt(this.publicKey, Buffer.from(value))
+    return this.serializeEcies(ecies)
   }
 
   private async decrypt(value: Uint8Array): Promise<Uint8Array> {
-    const ecies = deserializeEcies(value)
-    const result = await eccrypto.decrypt(this.privateKey, ecies)
+    const ecies = await this.deserializeEcies(value)
+    const result = await decrypt(this.privateKeyBytes, ecies)
     return Uint8Array.from(result)
   }
+
+  private async serializeEcies(data: Ecies): Promise<Uint8Array> {
+    // This will create and sign a `SignedEciesCiphertext` payload based on the provided data
+    const protoVal = await SignedEciesCiphertext.create(data, this.privateKey)
+    return protoVal.toBytes()
+  }
+
+  private async deserializeEcies(data: Uint8Array): Promise<Ecies> {
+    const protoVal = SignedEciesCiphertext.fromBytes(data)
+    // Verify the signature upon deserializing
+    if (!(await protoVal.verify(this.privateKey.publicKey))) {
+      throw new Error('signature validation failed')
+    }
+    const ecies = protoVal.ciphertext
+
+    return {
+      ciphertext: Buffer.from(ecies.ciphertext),
```

Comment:
The `ecies` library expects Buffers, and our protos deal in `Uint8Array`s. At some point we can decide whether we want to update `ecies` to take `Uint8Array` but for now I just convert everything.